src/index.js

```diff
@@ -61,31 +64,28 @@ const find = (haystack, searches, pathIn, parents, ctx) => {
       ctx.filterFn === undefined
       || ctx.filterFn(formatPath(pathIn, ctx), haystack, compiler.getMeta(searches, parents)) !== false
     ) {
-      if (ctx.callbackFn !== undefined) {
-        ctx.callbackFn(formatPath(pathIn, ctx), haystack, compiler.getMeta(searches, parents));
-      }
       result.push(formatPath(pathIn, ctx));
     }
   }
 
   return result;
 };
 
-module.exports = (needles, {
-  filterFn = undefined,
-  callbackFn = undefined,
-  breakFn = undefined,
-  joined = true,
-  escapePaths = true,
-  useArraySelector = true
-} = {}) => {
+module.exports = (needles, opts) => {
+  const ctx = Object.assign({
+    filterFn: undefined,
+    breakFn: undefined,
+    joined: true,
+    escapePaths: true,
+    useArraySelector: true
+  }, opts);
+  assert(Object.keys(ctx).length === 5, 'Unexpected Option provided!');
```

Comment:
I understand _why_ you're making this check, it's just unfortunate that we then lose code insight. Ah well, deficiency of the language I suppose

Reply:
Yeah it's stupid. But I much rather have this check than code insight